### PR TITLE
MAINT: Fixed about 100 unused imports, unused assignment warnings from scipy/optimize

### PR DESCRIPTION
**Pull request by pvanmulbregt**
Unused explicitly named imports have been removed.  Exceptions are:
  - imports in __init__.py;
  - imports used for side-effects.

#### Reference issue
Further work on gh-11529

#### What does this implement/fix?
Removes some unused imports, unused assignments to reduce Pyflakes warnings.

### CODE REVIEW

File: scipy/optimize/_trustregion_constr/tests/test_report.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-                           assert_array_equal, assert_array_less,
-                           assert_raises, assert_equal, assert_,
-                           run_module_suite, assert_allclose, assert_warns,
-                           dec)
+# import numpy as np
+# from numpy.testing import (TestCase, assert_array_almost_equal,
+#                            assert_array_equal, assert_array_less,
+#                            assert_raises, assert_equal, assert_,
+#                            run_module_suite, assert_allclose, assert_warns,
+#                            dec)
```
Comment (ev-br) [inline scipy/optimize/_trustregion_constr/tests/test_report.py]: Leftover?

**Comment by ev-br** [comment]
Could you also check that the number of public symbols stays the same? (Can be done with refguide_check, I guess).

This is just to make sure: a star import in `__init__.py` from a module X, which imports a name Z from module Y. Then Z ends up in the public API.

File: scipy/optimize/_trustregion_constr/tests/test_report.py (hunk 1 above)
Comment (pvanmulbregt) [reply scipy/optimize/_trustregion_constr/tests/test_report.py]: Yes, leftover.  Now removed.

**Comment by pvanmulbregt** [comment]
> Could you also check that the number of public symbols stays the same? (Can be done with refguide_check, I guess).

Is there any special invocation of `refguide_check.py` to check that, beyond the usual

```
$ python runtests.py -g  --refguide-check
```

it didn't report any errors. Is it sufficient?

**Comment by ev-br** [comment]
Likely, it's OK.
I'd still double-check the number of items in the `names_dict` in master and this PR,
https://github.com/scipy/scipy/blob/master/tools/refguide_check.py#L880

**Comment by andyfaff** [comment]
With the changes in this PR, and the related PR, it may make it possible to activate deactivated checks in the lint diff and pyflakes PEP8 tools.

**Comment by pvanmulbregt** [comment]
> I'd still double-check the number of items in the `names_dict` in master and this PR,

The output of 
```
python tools/refguide_check.py -v
```
didn't change.

**Comment by ev-br** [comment]
Thanks Paul

**Pull request merged**